docs/rest_api.md

```diff
@@ -2,12 +2,139 @@
 * `{"foo": <bar>, "baz": <quux>}` denotes a JSON object (in the body of 
   the request).
 
-Full Api spec:
+# Full Api spec:
 
-    user_create  <user_label> <password>
-    user_delete  <user_label>
-    [PUT]    /user/<user_label> {password=<password>}
-    [DELETE] /user/<user_label>
+## Users
+
+### user_create
+
+`PUT /user/<username>`
+
+Request body:
+
+    {
+        "password": <plaintext-password>
+    }
+
+Create a new user whose password is `<plaintext-password>`.
+
+Possible errors:
+
+* 409, if the user already exists
+
+### user_delete
+
+`DELETE /user/<username>`
+
+Delete the user whose username is `<username>`
+
+Possible errors:
+
+* 404, if the user does not exist
+
+## Networks
+
+### show_network
+
+`GET /network/<network>`
+
+View detailed information about `<network>`.
+
+The result contains the following information:
+
+* The name of the network
+* A description of legal channel identifiers for this network. This is a list
+  of channel identifiers, with possible wildcards. The format of these is
+  driver specific, see below.
+
+Response body (on success):
+
+    {
+        "name": <network>
+        "channels": <chanel-id-list>
+    }
+
+Possible errors:
+
+* 404, if the network does not exist.
+
+#### Channel Formats
+
+##### 802.1q VLAN based drivers
+
+Channel identifiers for the VLAN based drivers are one of:
+
+* `vlan/native`, to attach the network as the native (untagged) VLAN
+* `vlan/<vlan_id>` where `<vlan_id>` is a VLAN id number. This attaches
+   the network in tagged, mode, with the given VLAN id.
+
+Additionally, the `show_networks` api call may return the channel identifier
+`vlan/*`, which indicates that any VLAN-based channel id may be used.
+
+Where documentation specifies that the network driver should choose a
+default channel, the VLAN drivers choose `vlan/native`.
+
+### node_connect_network
+
+`POST /node/<node>/nic/<nic>/connect_network`
+
+Request body:
+
+    {
+        "network": <network>,
+        "channel": <channel> (Optional)
+    }
+
+Connect the network named `<network>` to `<nic>` on `<channel>`.
+
+`<channel>` should be a legal channel identifier specified by the output 
+of `show_network`, above. If `<channel>` is omitted, the driver will choose
+a default, typically some form of "untagged."
```

Comment:
Could we make the default behavior explicit (like "untagged")?

For something with potential security implications, I'd like to leave the default defined rather than implied.